src/Authentication/Authenticators/HmacSha256.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodeIgniter\Shield\Authentication\Authenticators;
+
+use CodeIgniter\HTTP\IncomingRequest;
+use CodeIgniter\I18n\Time;
+use CodeIgniter\Shield\Authentication\AuthenticationException;
+use CodeIgniter\Shield\Authentication\AuthenticatorInterface;
+use CodeIgniter\Shield\Entities\User;
+use CodeIgniter\Shield\Exceptions\InvalidArgumentException;
+use CodeIgniter\Shield\Models\TokenLoginModel;
+use CodeIgniter\Shield\Models\UserIdentityModel;
+use CodeIgniter\Shield\Models\UserModel;
+use CodeIgniter\Shield\Result;
+
+class HmacSha256 implements AuthenticatorInterface
+{
+    public const ID_TYPE_HMAC_TOKEN = 'hmac_sha256';
+
+    /**
+     * The persistence engine
+     */
+    protected UserModel $provider;
+
+    protected ?User $user = null;
+    protected TokenLoginModel $loginModel;
+
+    public function __construct(UserModel $provider)
+    {
+        $this->provider = $provider;
+
+        $this->loginModel = model(TokenLoginModel::class);
+    }
+
+    /**
+     * Attempts to authenticate a user with the given $credentials.
+     * Logs the user in with a successful check.
+     *
+     * @throws AuthenticationException
+     */
+    public function attempt(array $credentials): Result
+    {
+        /** @var IncomingRequest $request */
+        $request = service('request');
+
+        $ipAddress = $request->getIPAddress();
+        $userAgent = (string) $request->getUserAgent();
+
+        $result = $this->check($credentials);
+
+        if (! $result->isOK()) {
+            // Always record a login attempt, whether success or not.
```

Comment:
Do you really need to record all access?
See Authentication/Authenticators/JWT.php

Reply:
This is mimicking what is AuthorizeTokens is doing see line 54-55 in AccessTokens.php

Reply:
I think the implementation in `AuthorizeTokens` is a bug.
Nobody wants to record all accesses.

Reply:
@kenjis I finally had a chance to review this.  It looks like the issue is in two places.  The code block above has a comment that is either misplaced or inaccurate (depending on your perspective).    I will fix the comment.  Do we want to provide a config setting to allow for every login to be recorded?  There are some industries that do have a desire to record everything.  The questions are 1.) is this the right plugin to record all logins?  2.) Do we want the records to be in the database, or output to a log file (or both)?

Reply:
Oh, yes, the comment is misleading.

I think it should be implemented the same as JWT.
I plan to modify `AccessTokens` that way as well.

Reply:
Method is updated to more closely match the JWT logging authorization.  Of note I added a Config Class (named `AuthToken`) to contain the logging settings similar to the `AuthJWT` config class.